quantumlib/unitary
# Update copyright checker for pylint-3

The previous CI pylint check was ineffective.
Run it for real here and clean up some of the exposed lint.

* Use the latest pylint-3 for the lint CI check
* Copy pylint_copyright_checker from Cirq and adjust the copyright text
* Use the same pylint configuration as in Cirq, apart from line length
* pylint - fix singleton-comparison
* pylint - fix docstring-first-line-empty
* pylint - fix f-string-without-interpolation
* pylint - fix wrong-or-nonexistent-copyright-notice
* pylint - fix wrong-import-order
* pylint - fix unused-import
* Temporarily disable rules that fail in pylint-3 and mark them with TODO comments

Partially fixes #210
